### DISCUSSION TITLE
CapAwesome FCM Plugin vs official Capacitor Push Notification Plugin

### DISCUSSION THREAD

**Discussion by 5hee75** [capawesome-team/capacitor-firebase#336] [Q&A] [answered]
Forgive me if this is an obvious question and I'm just ignorant, but what would the motivation be behind using the CapAwesome FCM plugin over the official Capacitor Push Notifications plugin?  I've been using the official plugin for some time and don't seem to have any issues with it, albeit my use case is very basic.  Are there things the FCM plugin would provide that the official plugin would not?  Are there any features missing that the official plugin provides?

I do use Firebase pretty much across the board in my app (Auth, Firestore, Functions, Storage, Analytics), so if there'd be a benefit in switching to use the FCM plugin then I'd be open to it.

**Comment by robingenz** [comment, marked as answer]
Thanks for the question! 

The main differences are:
1. **Capacitor Push Notification != Firebase Messaging**: The Capacitor Push Notification Plugin is a more general plugin for integrating push notifications. It uses the Firebase SDK on Android but not on iOS. The Capacitor Firebase Cloud Messaging Plugin tries to support all features of the Firebase SDK. This includes, for example, the methods `subscribeToTopic(...)` and `unsubscribeFromTopic(...)`, which the Capacitor Push Notification Plugin does not offer.
2. **Web support:** Since Firebase offers a Web SDK, the Capacitor Firebase Cloud Messaging plugin also supports the web platform, unlike the Capacitor Push Notification plugin.
3. **No SDK version conflicts**: This monorepo was created mainly because version conflicts occurred when trying to use several Capacitor Firebase plugins from different maintainers at the same time, as the projects often did not use the same Firebase SDK version. This problem is solved by this monorepo. It ensures that the Capacitor Firebase Cloud Messaging plugin always remains compatible with all other Capacitor Firebase plugins.
